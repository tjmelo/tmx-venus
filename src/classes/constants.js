export const KEYS = {
  API: 'https://63f239144f17278c9a22d1cd.mockapi.io/api/v1/item',
};

export const ELEM = {
  ADD_ITEM: document.querySelector('.add-item'),
  ALL_ITEMS: document.querySelector('.all-items'),
};
