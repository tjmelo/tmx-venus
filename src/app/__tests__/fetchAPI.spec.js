import toFetchAPI from '../modules/fetchAPI';
import { KEYS } from '../constants';

const mockPostData = {
  method: 'POST',
  headers: { 'Content-Type': 'application/json' },
  body: 'some data',
};

jest.mock('../modules/fetchAPI');

describe('Should to call fetch API for render data', () => {
  it('Should fetch API data', async () => {
    const testToFetchAPI = jest.fn();
    testToFetchAPI.mockImplementation((argTest) => toFetchAPI(argTest));

    // do
    testToFetchAPI(KEYS.API, mockPostData);
    testToFetchAPI(KEYS.API);

    // then
    expect(testToFetchAPI).toHaveReturned();
    expect(testToFetchAPI).toHaveBeenCalledWith(KEYS.API);
    expect(testToFetchAPI).toHaveBeenCalledWith(KEYS.API, mockPostData);

    expect(testToFetchAPI).toHaveBeenCalledTimes(2);

    // Expects fidelity with scope of function
    expect(testToFetchAPI).toMatchSnapshot();
  });
});
