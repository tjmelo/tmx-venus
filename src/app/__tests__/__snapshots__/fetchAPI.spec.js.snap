// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should to call fetch API for render data Should fetch API data 1`] = `
[MockFunction] {
  "calls": [
    [
      "https://63f239144f17278c9a22d1cd.mockapi.io/api/v1/item",
      {
        "body": "some data",
        "headers": {
          "Content-Type": "application/json",
        },
        "method": "POST",
      },
    ],
    [
      "https://63f239144f17278c9a22d1cd.mockapi.io/api/v1/item",
    ],
  ],
  "results": [
    {
      "type": "return",
      "value": undefined,
    },
    {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;
